sdk/src/transferHookProvider.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  ExtensionType,
+  getExtensionData,
+  getExtraAccountMetaAddress,
+  getExtraAccountMetas,
+  getMetadataPointerState,
+  getMint,
+  getTransferHook,
+  Mint,
+  resolveExtraAccountMeta,
+  TOKEN_2022_PROGRAM_ID,
+} from '@solana/spl-token';
+import { unpack } from '@solana/spl-token-metadata';
+import { AccountMeta, Connection, PublicKey } from '@solana/web3.js';
+
+export interface TransferHookProvider {
```

Comment:
what we need is a central logic of `TransferHookProviderSelector`, and the goal can be minimizing the rpc requests